File: atcoder/abc200_e.py
# https://atcoder.jp/contests/abc200/tasks/abc200_e

def main():
    N, K = map(int, input().split())

    def c2(s):
        if s<0: return 0
        return s*(s-1)//2

    def f(s):
        res = c2(s-1)
        res -= c2(s-1-N)*3
        res += c2(s-1-N*2)*3
        res -= c2(s-1-N*3)
        return res

    def f2(s):
        l = max(1, s-N)
        r = min(N, s-1)
        if l>r: return 0
        return r-l+1

    for s in range(3, N*3+1):
        x = f(s)
        if K>x:
            K -= x
        else:
            for a in range(1, N+1):
                x = f2(s-a)
                if K>x:
                    K -= x
                else:
                    for b in range(1, N+1):
                        c = s-a-b
                        if c<=0 or c>N: continue
                        if K>1:
                            K -= 1
                        else:
                            print(a, b, c)
                            return

main()



# N, K = map(int, input().split())

# cum = 0
# for sum_ijk in range(3, 3*N+1):
#     comb = 0
#     for i in range(1, min(N,sum_ijk-2)+1):
#         comb += min(min(N, sum_ijk-(i-1)-2), max(0, 2*N-(sum_ijk-i)+1))
#     # print(sum_ijk, comb)
#     if cum+comb>=K:
#         break
#     cum += comb
# # print(K, cum, sum_ijk)

# cum2 = 0
# for i in range(max(1,sum_ijk-2*N), min(sum_ijk-2,N)+1):
#     if K-cum-cum2 > sum_ijk-i-1:
#         cum2 += sum_ijk-i-1
#     else:
#         j = (K-cum-cum2) + max(1, sum_ijk-i-N) - 1
#         cum2 += K-cum-cum2
#         break
# # print(K, cum+cum2)
# print(i, j, sum_ijk-i-j)



# from itertools import product
# N = 6
# cnt = [0]*(3*N+1)
# for i, j, k in product(range(1,N+1), repeat=3):
#     cnt[i+j+k] += 1

# print('sum', sum(cnt))
# for K in range(3, 3*N+1):
#     # ncr = (N+i-3-2)*(N+i-4-2)//2
#     comb = 0
#     for i in range(1, min(N,K-2)+1):
#         comb += min(min(N, K-(i-1)-2), max(0, 2*N-(K-i)+1))
#     print(K, cnt[K], comb)

File: atcoder/abc200_e.cpp
// https://atcoder.jp/contests/abc200/tasks/abc200_e
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
// #include <atcoder/all>
// using namespace atcoder;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define drep(i,n) for(int i = (n-1); i >= 0; i--)
#define all(v) (v).begin(),(v).end()
#define maxs(x,y) (x = max(x,y))
#define mins(x,y) (x = min(x,y))
template <class T> bool chmax(T &a, const T &b) { if (a < b) { a = b; return 1; } return 0; }
template <class T> bool chmin(T &a, const T &b) { if (b < a) { a = b; return 1; } return 0; }
template <class T> T gcd(T a, T b) { return b ? gcd(b, a % b) : a; }
template <class T> T lcm(T a, T b) { return a/gcd(a,b)*b; }
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;

int main() {
    cin.tie(nullptr);
    ios::sync_with_stdio(false);
	int n;
    ll k;
	cin >> n >> k;

    auto c2 = [&](ll s) {
        if (s < 0) return 0ll;
        return s*(s-1)/2;
    };

    auto f = [&](ll s) {
        ll res = c2(s-1);
        res -= c2(s-1-n)*3;
        res += c2(s-1-n*2)*3;
        res -= c2(s-1-n*3);
        return res;
    };

    auto f2 = [&](int s) {
        int l = max(1, s-n);
        int r = min(n, s-1);
        if (l > r) return 0;
        return r-l+1;
    };

    for (int s = 3; s <= n*3; ++s) {
        ll x = f(s);
        if (k > x) {
            k -= x;
        } else {
            for (int a = 1; a <= n; ++a) {
                x = f2(s-a);
                if (k > x) {
                    k -= x;
                } else {
                    for (int b = 1; b <= n; ++b) {
                        int c = s-a-b;
                        if (c <= 0 || c > n) continue;
                        if (k > 1) {
                            --k;
                        } else {
                            printf("%d %d %d\n", a, b, c);
                            return 0;
                        }
                    }
                }
            }

        }
    }
    return 0;
}
